activity-stream/send-to-device.html

```diff
@@ -0,0 +1,497 @@
+            case 'en-US':
+              check_phone = val.length === 10 && !isNaN(val);
+              break;
+            case 'de':
+              // afaict, german phone numbers can be between 2 and 11 digits
+              check_phone = val.length >= 2 && val.length <= 11 && !isNaN(val);
+              break;
+            // this case should never be hit, but good to have a fallback just in case
+            default:
+              check_phone = !isNaN(val);
+              break;
+          }
+          
+          if (check_email) {
+              return 'email';
+          } else if ('{{ include_sms }}' === 'True' && check_phone) {
+              return 'phone';
+          } else {
+              return false;
+          }
+        }
+  
+        // if fetch request fails, show the error message
+        function subscribeError() {
+          scene2.classList.add('hidden');
+          sceneError.classList.remove('hidden');
+        }
+    
+        // if fetch request is successful, show the success message
+        function subscribeOk() {
+          scene2.classList.add('hidden');
+          sceneOk.classList.remove('hidden');
+        }
+  
+        // both SMS and email subscriptions respond with the same JSON format
+        function performFetch(fetchRequest) {
+          fetch(fetchRequest).then(response => {
+            return response.json();
+          }).then(json => {
+            if (json.status === 'ok') {
+              subscribeOk();
+            } else {
+              subscribeError();
+            }
+          }).catch(err => {
+            subscribeError();
+          });
+        }
+  
+        function sendSMSToBasket() {
+          const formData = new FormData();
+          formData.append('mobile_number', input.value);
+          formData.append('msg_name', '{{ message_id_sms }}');
+          
+          const fetchConfig = {
+            body: formData,
+            method: 'post'
+          };
+  
+          const fetchRequest = new Request('https://basket.mozilla.org/news/subscribe_sms/', fetchConfig);
+  
+          performFetch(fetchRequest);
+        }
+  
+        function sendEmailToBasket() {
+          const formData = new FormData();
+          formData.append('email', input.value);
+          formData.append('newsletters', '{{ message_id_email }}');
+          // REVIEWER (Giorgos, most likely :)
+          // In a previous send to device snippet (/campaings/mobile/mobile-snipped-widget.html),
+          // the URL below has "{{ snippet_id }}" appended to the end. Is this value still available?
+          // It appears to be empty testing locally in simple_template_server, but maybe it's available
+          // in the real application?
+          formData.append('source_url', encodeURIComponent('https://snippets.mozilla.com/'));
+  
+          // Maybe change to the following?
+          // formData.append('source_url', encodeURIComponent('https://snippets.mozilla.com/{{ snippet_id }}'));
```

Comment:
I suggest `https://snippets.mozilla.com/show/{{ snippet_id }}` which displays a preview of the snippet. (if snippet is public)